lib/ansible/utils/display.py

```diff
@@ -570,14 +595,24 @@ def error(self, msg, wrap_text=True):
             self._errors[new_msg] = 1
 
     @staticmethod
-    def prompt(msg, private=False):
+    def prompt(msg: str, private: bool = False) -> str:
         if private:
             return getpass.getpass(msg)
         else:
             return input(msg)
 
-    def do_var_prompt(self, varname, private=True, prompt=None, encrypt=None, confirm=False, salt_size=None, salt=None, default=None, unsafe=None):
-
+    def do_var_prompt(
+        self,
+        varname: str,
+        private: bool = True,
+        prompt: str | None = None,
+        encrypt: str | None = None,
+        confirm: bool = False,
+        salt_size: int | None = None,
+        salt: str | None = None,
+        default: t.Any = None,
```

Comment:
Why not
```suggestion
        default: str | None = None,
```
?
Does it cause any problems?

Reply:
I don't think it causes any problems but the values passed to `do_var_prompt` method are coming from user and are run through templating so for native jinja it could just be anything and even for non-native there are multiple types that could be passed. Then there is the FIXME in the code below that the `vars_prompt` should be turned into a `class` type FA that would handle the types:

https://github.com/ansible/ansible/blob/fa5fe8a7db51cc5ff62689780db5296821708a19/lib/ansible/executor/playbook_executor.py#L135-L157

So this all is outside of the scope of this PR, so I guess for the time being we just expect `default` being a `str` `¯\_(ツ)_/¯`

Reply:
Fair enough. Future type coverage additions will likely reveal incompatibilities as more modules are covered with types…